references/optical_flow/train.py

```diff
@@ -115,21 +120,22 @@ def inner_loop(blob):
         inner_loop(blob)
         num_processed_samples += blob[0].shape[0]  # batch size
 
-    num_processed_samples = utils.reduce_across_processes(num_processed_samples)
+    if args.distributed:
+        num_processed_samples = utils.reduce_across_processes(num_processed_samples)
     print(
         f"Batch-processed {num_processed_samples} / {len(val_dataset)} samples. "
         "Going to process the remaining samples individually, if any."
     )
 
-    if args.rank == 0:  # we only need to process the rest on a single worker
+    if not args.distributed or args.rank == 0:  # we only need to process the rest on a single worker
         for i in range(num_processed_samples, len(val_dataset)):
             inner_loop(val_dataset[i])
 
     logger.synchronize_between_processes()
```

Comment:
Lines 124 to 134 (included) should actually be included in the `if args.distributed:` block. We only need to do these extra bits of work in distributed mode, because when in normal mode, the SequentialSampler will be processing all the samples properly. You should be able to remove the changes made to line 130 as well.